kubernetes/doc/create-options.md

```diff
@@ -0,0 +1,122 @@
+# Kubernetes createOptions Extensions
+
+IoT Edge on Kubernetes supports several experimental features to allow a user to assign additional parameters to pods that represents IoT Edge Modules.
+
+## Experimental features
+
+All features described in this document are experimental. To enable them, the following environment variable need to be set for the edgeAgent (make note of the double underscores):
+
+| Environment Variable Name                   | value  |
+|---------------------------------------------|--------|
+| `ExperimentalFeatures__Enabled`             | `true` |
+| `ExperimentalFeatures__EnableK8SExtensions` | `true` |
+
+## Create Options
+
+We added CreateOptions for experimental features on Kubernetes. These options will be ignored until both `ExperimentalFeatures__Enabled` and `ExperimentalFeatures__EnableK8SExtensions` are not set to `true`.
+
+```json
+{
+  "k8s-experimental": {
+    "volumes": [{...}],
+    "resources": [{...}],
+    "nodeSelector": {...}
+  }
+}
+```
+
+## Volumes
+
+EdgeAgent allows to mount existing volumes in the namespace from different sources e.g. `ConfigMap` as a pod `Volume`.
+
+### Create Options
+
+A `volumes` section of config used to describe how a `ConfigMap` or any other volume source existing in a namespace will be mounted into module container. The value of this config section is an array of pairs `volume` and `volumeMount`. A `volume` part describes how a specified volume source is mounted to `pod` and `volumeMounts` describes a mounting of a `volume` within a module main container. See [Volumes](https://kubernetes.io/docs/concepts/storage/volumes/) and [PodSpec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#podspec-v1-core) reference for more information.
+
+`EdgeAgent` doesn't do any translations or interpretations of values but simply assigns values to the `pod` and `container` specs. The description of exact structure can be found here for [Volume](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#volume-v1-core) and for [VolumeMount](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#volumemount-v1-core).
+
+```json
+{
+  "k8s-experimental": {
+    "volumes": [
+      {
+        "volume": {
+          "name": "ModuleA",
+          "configMap": {
+            "optional": "true",
+            "defaultMode": 420,
+            "items": [{
+                "key": "config-file",
+                "path": "config.yaml",
+                "mode": 420
+            }],
+            "name": "module-config"
+          }
+        },
+        "volumeMounts": [
+          {
+            "name": "module-config",
+            "mountPath": "/etc/module/config.yaml",
+            "mountPropagation": "None",
+            "readOnly": "true",
+            "subPath": "" 
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+## CPU, Memory and Device Resources
+
+EdgeAgent allows to specify Compute resources for a module container.
+
```

Comment:
Worth adding the point about what EdgeAgent does with HostConfig CPU and memory limits. Because the CPU memory limits are pretty much passthrough k8s via limits below